boards/sim/sim/sim/configs/nsh2/Make.defs

```diff
@@ -82,6 +82,18 @@ else
   LDMODULEFLAGS += -T $(TOPDIR)/libs/libc/modlib/gnu-elf.ld
 endif
 
+# NuttX modules are ELF binaries.
+# Non-ELF platforms like macOS need to use a separate ELF toolchain.
+ifeq ($(CONFIG_HOST_MACOS),y)
+  # eg. brew install x86_64-elf-gcc
+  MODULECC = x86_64-elf-gcc
```

Comment:
Should we change to:
MODULECC ?= x86_64-elf-gcc
so the user have a chance to overwrite the toolchain?

Reply:
it doesn't work as here we are overriding the default value from Config.mk.

a user can override it by "make MODULECC=foo" even with the current code.